mesh/submesh/psubmesh.cpp

```diff
@@ -235,10 +235,11 @@ ParSubMesh::ParSubMesh(const ParMesh &parent, SubMesh::From from,
 
    // Add boundaries
    {
-      int num_of_faces_or_edges =
-         (Dim == 3) ? NumOfFaces :
-         ((Dim == 2) ? NumOfEdges : NumOfVertices);
-      Array<int> &be2face = (Dim == 2) ? be_to_edge : be_to_face;
+      int num_codim_1 = 0;
+      if (Dim == 1) { num_codim_1 = NumOfVertices; }
+      else if (Dim == 2) { num_codim_1 = NumOfEdges; }
+      else if (Dim == 3) { num_codim_1 = NumOfFaces; }
+      else { MFEM_ABORT("Invalid dimension."); }
```

Comment:
If you'd like, you can bind this to a `const`:
```
const int num_codim_1 = [this]()
{
      if (Dim == 1) { return NumOfVertices; }
      else if (Dim == 2) { return NumOfEdges; }
      else if (Dim == 3) { return NumOfFaces; }
      else { MFEM_ABORT("Invalid dimension."); }
      return -1;
}();
```

Reply:
Yeah, I sometimes use this idiom (["IIFE"](https://rigtorp.se/iife/)). It's good because of const, but it adds some "syntax noise" to the code (wouldn't it be great if C++ had "if-expressions"?). I'm fine either way.

Reply:
Yeah I guess I'm just very used to reading them now and like the const, I barely notice. Not worth holding the pr for, it's an aesthetic choice rather than functional.

Reply:
Done in 11aa88c5e